docs/spec_draft.md

```diff
@@ -388,6 +389,50 @@ dimensions `k` in `operand`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.case
+
+### Semantics
+
+Returns the result of exactly 1 function from `branches` depending on the value
+of `index`. Formally, if $0 \le$ `index` $\lt$ `N-1`, result of executing
+`branches[index]` is returned, else, result of executing `branches[N-1]` is
+returned.
+
+### Inputs
+
+| Name       | Type                                         |
+|------------|----------------------------------------------|
+| `index`    | 1-dimensional tensor constant of type `si32` |
+| `branches` | a variable sized array of type `function`    |
+
+### Outputs
+
+| Name     | Type               |
+|----------|--------------------|
+| `result` | any supported type |
+
+### Constraints
+
+  * (C1) `branches` have atleast one function.
+  * (C2) All functions in `branches` take 0 arguments.
+  * (C3) Return type of all functions in `branches` are same.
```

Comment:
This also needs "(C4) For all `i`, `type(results[i]) = type(branches[0]).outputs[i]`". Related discussion: https://github.com/openxla/stablehlo/pull/325/files#r996113199.

Reply:
`result[i]`

Reply:
Since result is a variadic tensor, it could me made into results, but singular is also ok grammatically.
Let me know if I should change both places to `results`.